monad-logger/Control/Monad/Logger.hs

```diff
@@ -584,19 +591,67 @@ defaultLogStr loc src level msg =
             LS s -> encodeUtf8 $ pack s
             LB b -> b
         , S8.pack " @("
-        , encodeUtf8 $ pack fileLocStr
+        , encodeUtf8 $ pack (fileLocStr loc)
         , S8.pack ")\n"
         ]
 #endif
+
+-- taken from file-location package
+-- turn the TH Loc loaction information into a human readable string
+-- leaving out the loc_end parameter
+fileLocStr :: Loc -> String
+fileLocStr loc = (loc_package loc) ++ ':' : (loc_module loc) ++
+  ' ' : (loc_filename loc) ++ ':' : (line loc) ++ ':' : (char loc)
   where
-    -- taken from file-location package
-    -- turn the TH Loc loaction information into a human readable string
-    -- leaving out the loc_end parameter
-    fileLocStr = (loc_package loc) ++ ':' : (loc_module loc) ++
-      ' ' : (loc_filename loc) ++ ':' : (line loc) ++ ':' : (char loc)
-      where
-        line = show . fst . loc_start
-        char = show . snd . loc_start
+    line = show . fst . loc_start
+    char = show . snd . loc_start
+
+defaultTimedLogStr :: Loc
+              -> LogSource
+              -> LogLevel
+              -> LogStr
+              -> FormattedTime
+#if MIN_VERSION_fast_logger(0, 2, 0)
+              -> LogStr
+#else
+              -> S8.ByteString
+#endif
+defaultTimedLogStr loc src level msg time =
+#if MIN_VERSION_fast_logger(0, 2, 0)
```

Comment:
I think this PR depends on new changes in the fast-logger library. If so: it would be better to update the lower bound in the .cabal file and then get rid of the unnecessary MIN_VERSION CPP.